src/elf/sym.rs

```diff
@@ -320,114 +320,112 @@ pub mod sym64 {
     elf_sym_std_impl!(u64);
 }
 
-if_alloc! {
-    use scroll::ctx;
-    use scroll::ctx::SizeWith;
-    use core::fmt::{self, Debug};
-    use core::result;
-    use crate::container::{Ctx, Container};
-    use crate::error::Result;
-    use crate::alloc::vec::Vec;
-
-    #[derive(Default, PartialEq, Clone)]
-    /// A unified Sym definition - convertable to and from 32-bit and 64-bit variants
-    pub struct Sym {
-        pub st_name:     usize,
-        pub st_info:     u8,
-        pub st_other:    u8,
-        pub st_shndx:    usize,
-        pub st_value:    u64,
-        pub st_size:     u64,
-    }
+use scroll::ctx;
+use scroll::ctx::SizeWith;
+use core::fmt;
+use crate::container::{Ctx, Container};
+#[cfg(feature = "alloc")]
```

Comment:
I don't like it too. But since `gnu_hash` returns a `elf::sym::Sym`. Also
there are no cfgs around gnu_hash, so `elf::sym::Sym` shouldn't have that too.